api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -64,74 +55,49 @@ public boolean deleteInstitution(final String shortName) {
   public Optional<Institution> updateInstitution(
       final String shortName, final Institution institutionToUpdate) {
     return getDbInstitution(shortName)
-        .map(dbInst -> toModelClass(saveInstitution(institutionToUpdate, dbInst)));
+        .map(dbInst -> toModel(institutionDao.save(updateDbObject(dbInst, institutionToUpdate))));
   }
 
   private Optional<DbInstitution> getDbInstitution(String shortName) {
     return institutionDao.findOneByShortName(shortName);
   }
 
-  private DbInstitution saveInstitution(final Institution modelClass, final DbInstitution dbClass) {
-    dbClass.setShortName(modelClass.getShortName());
-    dbClass.setDisplayName(modelClass.getDisplayName());
-    dbClass.setOrganizationTypeEnum(
-        DbStorageEnums.organizationTypeToStorage(modelClass.getOrganizationTypeEnum()));
-    dbClass.setOrganizationTypeOtherText(modelClass.getOrganizationTypeOtherText());
-
-    return saveInstitutionEmailPatterns(modelClass, institutionDao.save(dbClass));
+  private DbInstitution newDbObject(final Institution modelObject) {
+    return updateDbObject(new DbInstitution(), modelObject);
   }
 
-  private DbInstitution saveInstitutionEmailPatterns(
-      final Institution modelClass, final DbInstitution dbClass) {
-
-    institutionEmailDomainDao.deleteAllByInstitution(dbClass);
-    Optional.ofNullable(modelClass.getEmailDomains())
-        .ifPresent(
-            domains ->
-                dbClass.setEmailDomains(
-                    institutionEmailDomainDao.save(
-                        domains.stream()
-                            .map(domain -> new DbInstitutionEmailDomain(dbClass, domain))
-                            .collect(Collectors.toSet()))));
-
-    institutionEmailAddressDao.deleteAllByInstitution(dbClass);
-    Optional.ofNullable(modelClass.getEmailAddresses())
-        .ifPresent(
-            addresses ->
-                dbClass.setEmailAddresses(
-                    institutionEmailAddressDao.save(
-                        addresses.stream()
-                            .map(address -> new DbInstitutionEmailAddress(dbClass, address))
-                            .collect(Collectors.toSet()))));
-
-    return institutionDao.save(dbClass);
+  private DbInstitution updateDbObject(
+      final DbInstitution dbObject, final Institution modelObject) {
+    return dbObject
+        .setShortName(modelObject.getShortName())
+        .setDisplayName(modelObject.getDisplayName())
+        .setOrganizationTypeEnum(modelObject.getOrganizationTypeEnum())
+        .setOrganizationTypeOtherText(modelObject.getOrganizationTypeOtherText())
+        .setEmailDomains(
```

Comment:
I like chained stream statements, but I have trouble reading nested streams. What would you think about making little helper methods to process the email domains and addresses?

Reply:
That is coming tomorrow when I convert to MapStruct